### ISSUE TITLE
C++ - Incorrect function info generated for class constructors

### ISSUE THREAD

**Issue by audreyseo** [nikhilkalige/docblockr#273]
Attempting to document C++ constructors results in either no information or erroneously selected information from below the constructors being inserted into the comment body. For example, it will use the next closest object initialization:

```C++
/**
 * [baz description]
 *
 * @param  bar [description]
 * @return     [description]
 */
Foo::Foo(std::string bar) {
  // Code [no object initializations]...
  // The object whose initialization it mistakes for a method definition
  std::ifstream baz(bar);
  // More code...
}
```

Or, when there aren't any object initializations in the constructor, if there is a function definition after the constructor that doesn't have a doc comment, it will use the information from that:

```C++
/**
 * [Foo::bar description]
 *
 * @param <args...>
 */
Foo::Foo(<constructor args...>) {
  // constructor code...
}

void Foo::bar(<args...>) {
  // class method code...
}
```

**Comment by MoritzKn** [comment]
Thanks for reporting and immediate addressing the issue yourself! I'm currently reviewing your PR.

**Comment by MoritzKn** [comment]
Fix in #272 by @audreyseo and shipped with `v0.10.5`
